cdap-kafka/src/main/java/co/cask/cdap/kafka/run/KafkaServerMain.java

```diff
@@ -105,8 +101,15 @@ public void init(String[] args) {
     int brokerId = generateBrokerId(address);
     LOG.info(String.format("Initializing server with broker id %d", brokerId));
 
-    kafkaProperties = generateKafkaConfig(brokerId, zkConnectStr, hostname, port, numPartitions,
-                                          replicationFactor, logDir);
+    kafkaProperties.setProperty("broker.id", Integer.toString(brokerId));
+    if (hostname != null) {
+      kafkaProperties.setProperty("host.name", hostname);
+    }
+    kafkaProperties.setProperty("port", Integer.toString(port));
```

Comment:
are you sure 60-61 can be removed? what about everything that is done in 63-91??